### PR TITLE
Support to parse multiple files at the same time

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -30,7 +59,7 @@ func ParseFile(path string) (*GoFile, error) {
 		Uses:  make(map[*ast.Ident]types.Object),
 	}
 
-	_, err = conf.Check(file.Name.Name, fset, []*ast.File{file}, info)
+	_, err = conf.Check(file.Name.Name, fset, files, info)
```
Comment (zpatrick) [inline parser.go]:
```
if _, err := conf.Chec(...); err != nil {
    return nil, err
}
```

File: parser.go [hunk 2]
```diff
@@ -10,18 +10,47 @@ import (
...
 
-func ParseFile(path string) (*GoFile, error) {
-	source, err := ioutil.ReadFile(path)
-	if err != nil {
-		return nil, err
+// ParseFiles parses files at the same time
+func ParseFiles(paths []string) ([]*GoFile, error) {
+	var files []*ast.File
+	var fsets []*token.FileSet
+	for _, p := range paths {
```
Comment (zpatrick) [inline parser.go]: couldn't this function be:
```go
func ParseFiles(paths []string) ([]*GoFile, error) {
    goFiles := make([]*GoFile, len(paths))
    for i, path := range paths {
        goFile, err := ParseSingleFile(p)
        if err != nil {
            return nil, err
        }

        goFiles[i] = goFile
    }

    return goFiles, nil
}
```

**Review by zpatrick — CHANGES_REQUESTED** [review]
Thanks for the PR! Couple suggestions/questions

File: parser.go (hunk 1 above)
Comment (markya0616) [reply parser.go]: OK

File: parser.go (hunk 2 above)
Comment (markya0616) [reply parser.go]: Updated it. In golang, we can define structs in different files in the same package. We need to pass more `files []*ast.File` in `parseFile` to check.

**Review by zpatrick — APPROVED** [review]

**Pull request merged**